### PR TITLE
TypeError when GitHub URL is not defined #52

### PR DESCRIPTION
**Pull request by zubinbhasin**
#52

### CODE REVIEW

**Comment by helio-frota** [comment]
@zubinbhasin thanks for the PR.
Can you please write a unit test for when Github URL is not found?
You can take a look here for similar examples https://github.com/nodeshift/npcheck/blob/main/test/plugins/archive.test.js#L80

File: src/plugins/archive.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
-const { createError } = require('../lib/result');
-const { stringBuilder, success, failure } = require('../lib/format');
+const { createError, createWarning } = require('../lib/result');
+const { stringBuilder, success, failure, warning } = require('../lib/format');
 const { fetchGithub } = require('../lib/fetch');
 
 const archivePlugin = async (pkg, _, options) => {
+  if (!pkg.repository.url) {
+    warning(output.get());
+    return createWarning(`No Github url found for the ${pkg.name} module`);
```
Comment (aalykiot) [inline src/plugins/archive.js:8]:
```suggestion
    return createWarning(`The module "${pkg.name}" does not specify a GitHub URL.`);
```

**Comment by zubinbhasin** [comment]
> @zubinbhasin thanks for the PR.
> Can you please write a unit test for when Github URL is not found?
> You can take a look here for similar examples https://github.com/nodeshift/npcheck/blob/main/test/plugins/archive.test.js#L80

I have added the unit test and made the changes according to the suggestions by @aalykiot

**Review by helio-frota — APPROVED** [review]

**Review by richardlau — APPROVED** [review]

File: src/plugins/archive.js [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+const { stringBuilder, success, failure, warning } = require('../lib/format');
 const { fetchGithub } = require('../lib/fetch');
 
 const archivePlugin = async (pkg, _, options) => {
+
+  const output = stringBuilder(
+    'Checking if github repository is archived'
+  ).withPadding(65);
+
+  if (!pkg.repository.url) {
```
Comment (aalykiot) [inline src/plugins/archive.js:11]: If the `repository` field is also undefined then the if statement will throw a TypeError.

```suggestion
  if (!pkg?.repository?.url) {
```

File: test/plugins/archive.test.js [hunk 3]
```diff
@@ -95,3 +96,22 @@ it('should log failure message when module is archived', async () => {
 
   expect(format.failure).toHaveBeenCalled();
 });
+
+it('should log warning message when module does not specify its GitHub repository', async () => {
+  // mocking http request to GitHub
+  network.fetchGithub.mockImplementation(() => {
```
Comment (aalykiot) [inline test/plugins/archive.test.js:102]: You don't have to mock the network request here since you'll never reach it.

**Comment by aalykiot** [comment]
@zubinbhasin I would just writing one test with `const pkg = {};` and another with `const pkg = { repository: {} };` so we can cover all the edge cases 🙂

**Comment by zubinbhasin** [comment]
> @zubinbhasin I would suggest writing one test with `const pkg = {};` and another with `const pkg = { repository: {} };` so we can cover all the edge cases 🙂

@aalykiot I've covered both of these cases, also removed the mock calls and added optional chaining snippet as mentioned above.

**Review by aalykiot — APPROVED** [review]
@zubinbhasin LGTM, can you please rebase from `main` so we can merge your PR ??

**Comment by zubinbhasin** [comment]
> @zubinbhasin LGTM, can you please rebase from `main` so we can merge your PR ??

@aalykiot I've done the rebase but coveralls seems to be failing now.

**Comment by helio-frota** [comment]
> @zubinbhasin LGTM, can you please rebase from main so we can merge your PR ??

> @aalykiot I've done the rebase but coveralls seems to be failing now.

Don't need to worry about that coveralls issue :+1:
